src/classes/OPP_OpportunityNaming_TEST.cls

```diff
@@ -276,4 +278,207 @@ private class OPP_OpportunityNaming_TEST
         System.assertEquals(1, errors.size(), 'An error record should be created.');
 
     }
+
+    /*******************************************************************************************************
+    * @description Verify if opportunity naming format works when creating opportunity using a user with
+    * language other than English.
+    ********************************************************************************************************/
+    @isTest
+    private static void shouldFormatOppNameWhenInsertingForNonEnglishUser() {
+
+        Opportunity_Naming_Settings__c ons = new Opportunity_Naming_Settings__c(
+            Name = 'foo',
+            Opportunity_Name_Format__c = '{!Contact.Name} {!CloseDate} ',
+            Attribution__c = STG_PanelOppNaming_CTRL.oppAttributionNames.Both.name()
+        );
+
+        insert ons;
+
+        Account acc = new Account(Name='accname');
+        insert acc;
+        Contact con = new Contact(LastName='conname');
+        insert con;
+
+        Profile standardProfile = [SELECT Id FROM Profile WHERE Name='Standard User'];
+        //spanish language users to use for opp name formatting
+        User spanishLanguageUser = buildSpanishUser(standardProfile);
+
+        System.runAs(spanishLanguageUser) {
+            Opportunity opp = new Opportunity(
+            AccountId = acc.id,
+            StageName = UTIL_UnitTestData_TEST.getClosedWonStage(),
+            Name='temp',
+            Amount=8,
+            CloseDate = CLOSE_DATE,
+            npe01__Contact_Id_for_Role__c = con.id
+            );
+
+            insert opp;
+
+            List<Opportunity> queryOpp = [SELECT Name FROM Opportunity WHERE Id = :opp.id];
+
+            String oppName = 'conname';
+            DateTime closeDatetime = Datetime.newInstance(CLOSE_DATE.year(), CLOSE_DATE.month(), CLOSE_DATE.day());
+            oppName +=  ' ' + closeDatetime.format('yyyy.MM.dd');
```

Comment:
There is a repetition of the `closeDatetime` construct throughout tests, let's add the value or its format into the constant.